src/rt/model.c

```diff
@@ -810,6 +810,11 @@ const void *signal_last_value(rt_signal_t *s)
    return s->shared.data + s->shared.size;
 }
 
+uint32_t signal_width(rt_signal_t *s)
+{
+   return s->nexus.width;
```

Comment:
```suggestion
   return s->shared.size / s->nexus.size;
```

There's a difference if sub-elements of a composite signal have to be tracked independently. For example if you have two processes:

```vhdl
signal x : std_logic_vector(2 downto 0);
...
p1: x(2 downto 1) <= a;
p2: x(0) <= b;
```

Then the `rt_signal_t` for `x` has two `rt_nexus_t` the first with width 2 and the second with width 1 but the overall signal width should be 3. (`shared.size` is in bytes rather than elements because it's more convenient for the generated code.)

Reply:
OK, thanks for the explanation. I couldn't figure out how width differed from that calculation.